internal/pkg/dl/policies.go

```diff
@@ -0,0 +1,82 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package dl
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fleet/internal/pkg/bulk"
+	"fleet/internal/pkg/model"
+	"fmt"
+	"sync"
+
+	"fleet/internal/pkg/dsl"
+)
+
+var (
+	tmplQueryLatestPolicies     []byte
+	initQueryLatestPoliciesOnce sync.Once
+)
+
+var ErrPolicyLeaderNotFound = errors.New("policy has no leader")
+
+func prepareQueryLatestPolicies() []byte {
+	root := dsl.NewRoot()
+	root.Size(0)
+	policyId := root.Aggs().Agg(FieldPolicyId)
+	policyId.Terms("field", FieldPolicyId, nil)
+	revisionIdx := policyId.Aggs().Agg(FieldRevisionIdx).TopHits()
+	revisionIdx.Size(1)
+	rSort := revisionIdx.Sort()
+	rSort.SortOrder(FieldRevisionIdx, dsl.SortDescend)
+	rSort.SortOrder(FieldCoordinatorIdx, dsl.SortDescend)
+	return root.MustMarshalJSON()
+}
+
+// QueryLatestPolices gets the latest revision for a policy
+func QueryLatestPolicies(ctx context.Context, bulker bulk.Bulk, opt ...Option) ([]model.Policy, error) {
+	initQueryLatestPoliciesOnce.Do(func() {
+		tmplQueryLatestPolicies = prepareQueryLatestPolicies()
+	})
```

Comment:
nit:
I moved all the queries preparations into package inited vars in the last PRs, cleaner and should fail sooner.
I can take care of it later when it's merged to make it more consistent